cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -242,27 +469,39 @@ std::unique_ptr<column> drop_list_duplicates(lists_column_view const& lists_colu
     CUDF_FAIL("Nested types are not supported in drop_list_duplicates.");
   }
 
-  // Call segmented sort on the list elements and store them in a temporary column sorted_list
-  auto const sorted_lists =
-    detail::sort_lists(lists_column, order::ASCENDING, null_order::AFTER, stream);
-
   // Flatten all entries (depth = 1) of the lists column
-  auto const all_lists_entries = lists_column_view(sorted_lists->view()).get_sliced_child(stream);
+  auto const lists_entries = lists_column.get_sliced_child(stream);
+
+  // sorted_lists will store the results of the original lists after calling segmented_sort
+  auto const sorted_lists = [&]() {
+    // If the column contains lists of floating point data type and NaNs are considered as equal, we
+    // need to replace -NaN by NaN before sorting
+    auto const has_negative_nan =
+      type_dispatcher(lists_entries.type(), detail::has_negative_nans_fn{}, lists_entries, stream);
```

Comment:
You only want to do this check if `nans_equal == ALL_EQUAL`, right?

Reply:
Right, I forgot to check that too. Thanks.